clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/validators/DeployAppengineDescriptionValidator.groovy

```diff
@@ -40,15 +40,23 @@ class DeployAppengineDescriptionValidator extends DescriptionValidator<DeployApp
     }
 
     boolean isContainerDeployment = description.containerImageUrl?.trim()
+    boolean usesGcs = description.repositoryUrl?.startsWith("gs://")
 
-    if (!isContainerDeployment && !description.repositoryUrl.startsWith("gs://")) {
-      if (!helper.validateGitCredentials(description.credentials.gitCredentials,
-                                         description.gitCredentialType,
-                                         description.credentials.name,
-                                         "gitCredentialType")) {
-         return
+    if (!description.artifact) {
+      if (isContainerDeployment) {
+        helper.validateNotEmpty(description.containerImageUrl, "containerImageUrl")
+      } else {
+        if (!usesGcs) {
```

Comment:
Seems like this could be an `else if` and `else` in the conditional one level up.

Reply:
Is this roughly what you mean?

```groovy
if (!description.artifact) {
  if (isContainerDeployment) {
    helper.validateNotEmpty(description.containerImageUrl, "containerImageUrl")
  } else if (!usesGcs) {
    if (!helper.validateGitCredentials(/* stuff */)) {
      return
    }
    helper.validateNotEmpty(description.branch, "branch")
    helper.validateNotEmpty(description.repositoryUrl, "repositoryUrl")
  } else {
    helper.validateNotEmpty(description.repositoryUrl, "repositoryUrl")
  }
}
```

We end up duplicating the `validateNotEmpty` of `repositoryUrl` in this version.  I think I might be misreading your comment here, what do you expect the code to look like?

Reply:
I see -- didn't notice that, I thought that statement would be in the final `else` block. Thanks for clarifying, good as is.